egs/sre08/v1/sid/nnet3/xvector/get_egs_v2.sh

```diff
@@ -0,0 +1,228 @@
+    utils/fix_data_dir.sh $temp_dir/$subdir
+    utils/sym2int.pl -f 2 $data/spk2int $temp_dir/$subdir/utt2spk \
+      > $temp_dir/$subdir/utt2int
+  done
+
+  echo "$0: combine and shuffle (valid)"
+  #combine feats.scp and shuffle
+  mkdir -p $temp_dir/combine
+  data_combine=$temp_dir/combine
+
+  utils/combine_data.sh $data_combine $temp_dir/data_*
+  utils/shuffle_list.pl $data_combine/feats.scp > $data_combine/shuffled_feats.scp
+  utils/sym2int.pl -f 2 ${data}/spk2int ${data_combine}/utt2spk \
+    > ${data_combine}/utt2int
+
+  feat_scps=$(for n in `seq $kinds_of_length_valid`; do echo $data_combine/shuffled_feats.${n}.scp; done)
+  utils/split_scp.pl ${data_combine}/shuffled_feats.scp $feat_scps
+
+  echo "$0: dump egs (valid)"
+  #dump egs
+  $cmd --max-jobs-run $nj JOB=1:$kinds_of_length_valid $dir/log/dump_egs_valid.JOB.log \
+    nnet3-xvector-get-egs-seg \
+      --num-pdfs=$num_pdfs \
+      $temp_dir/combine/utt2int \
+      scp:$temp_dir/combine/shuffled_feats.JOB.scp \
+      ark,scp:$dir/valid_egs.JOB.ark,$dir/valid_egs.JOB.scp
+  cat $dir/valid_egs.*.scp > $dir/valid_diagnostic.scp
+fi
+
+temp_dir=$dir/train_subset/temp
+if [ $stage -le 5 ]; then
+  echo "$0: Dealing with train_diagnostic set"
+  #Generate data_li/feat.scp and utt2spk
+  $cmd $dir/log/allocate_examples_diagnostic.log \
+    sid/nnet3/xvector/allocate_egs_seg_v2.py \
+      --min-frames-per-chunk=$min_frames_per_chunk \
+      --max-frames-per-chunk=$max_frames_per_chunk \
+      --kinds-of-length=$kinds_of_length_valid \
+      --generate-validate=true \
+      --randomize-chunk-length=false \
+      --data-dir=$data/train_subset \
+      --output-dir=$temp_dir || exit 1
+  #Generate data_li/spk2utt and utt2label
+  for subdir in `dir $temp_dir`; do
+    utils/fix_data_dir.sh $temp_dir/$subdir
+    utils/sym2int.pl -f 2 $data/spk2int $temp_dir/$subdir/utt2spk \
+      > $temp_dir/$subdir/utt2int
+  done
+
+  echo "$0: combine and shuffle (diagnostic)"
+  #combine feats.scp and shuffle
+  mkdir -p $temp_dir/combine
+  data_combine=$temp_dir/combine
+
+  utils/combine_data.sh $data_combine $temp_dir/data_*
+  utils/shuffle_list.pl $data_combine/feats.scp > $data_combine/shuffled_feats.scp
+  utils/sym2int.pl -f 2 ${data}/spk2int ${data_combine}/utt2spk \
+    > ${data_combine}/utt2int
+
+  feat_scps=$(for n in `seq $kinds_of_length_valid`; do echo $data_combine/shuffled_feats.${n}.scp; done)
+  utils/split_scp.pl ${data_combine}/shuffled_feats.scp $feat_scps
+
+  echo "$0: dump egs (diagnostic)"
+  #dump egs
+  $train_cmd --max-jobs-run $nj JOB=1:$kinds_of_length_valid $dir/log/dump_egs_diagnostic.JOB.log \
+    nnet3-xvector-get-egs-seg \
+      --num-pdfs=$num_pdfs \
+      $temp_dir/combine/utt2int \
+      scp:$temp_dir/combine/shuffled_feats.JOB.scp \
+      ark,scp:$dir/train_diagnostic_egs.JOB.ark,$dir/train_diagnostic_egs.JOB.scp
+  cat $dir/train_diagnostic_egs.*.scp > $dir/train_diagnostic.scp
+  ln -s $dir/train_diagnostic.scp combine.scp
+fi
```

Comment:
I am not sure, why you removed nnet3-shuffle-egs to shuffle order of archives on disc?

Reply:
Previously, after generate some new feats.scp files in different directories. We combine them and we have already used "utils/shuffle_list.pl $data_combine/feats.scp > $data_combine/shuffled_feats.scp" to get a disorder list. So the egs in each archive has already been disordered.